Fixes from warning generation in Clang(windows) and MSVC

aaye opened the pull request:
### Description of Change(s)
 - Fix for the use of "long" type in printf format strings when paired with a size_t argument. Change from "l" to "z".
 - Fix for testing an array variable for null (which would always fail). Changed to testing the expected length of the array (larger than zero)
 - Added a couple of pragmas for disabling warnings that I will need later.

### Fixes Issue(s)
- Compile warnings

<!--
Please follow the Contributing and Building guidelines to run tests against your
change. Place an X in the box if tests are run and are all tests passing.
-->
- [x] I have verified that all unit tests pass with the proposed changes
<!-- 
Place an X in the box if you have submitted a signed Contributor License Agreement.
A signed CLA must be received before pull requests can be merged.
For instructions, see: http://openusd.org/release/contributing_to_usd.html
-->
- [x] I have submitted a signed Contributor License Agreement

jesschimein commented:
Filed as internal issue #USD-8691

aaye commented:
I seem to be hitting the 2-hour limit on the windows compile across my changes. Anything that can be done about that?

spiffmon commented:
We’re looking into it ($$), but don’t expect it will be resolved in the
immediate future.

On Fri, Sep 15, 2023 at 3:49 PM Andrew Aye ***@***.***> wrote:

> I seem to be hitting the 2-hour limit on the windows compile across my
> changes. Anything that can be done about that?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PixarAnimationStudios/OpenUSD/pull/2677#issuecomment-1721973161>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABPOU2HPHN3ZGAIIPE7A7KDX2TLOFANCNFSM6AAAAAA4VJIXDI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
--spiffiPhone

aaye commented:
Thanks! The rest all got through so not that big of a hit in terms of value for time spent. Will see if there is anything I can do on my end.

aaye commented:
Got pinged recently on an issue with PCH on Windows builds etc. I know its been a year, updated my changes to the most recent and updated the branch.

aaye committed:
MSVC and Clang fixes for compile warnings and PCH

Removed unused member variables
Printf format for a size_t should be zu (since the size of the variable is by definition capable of being different integer lengths).
Fixed some API tags that were causing a conflict in export/import pairing
Added some missing override keywords
Added some ARCH_PRAGMA warning disables around some code that was causing Windows Clang (clang version 19.1.3, Target: x86_64-pc-windows-msvc, Thread model: posix) to throw warnings.
Fixed windows linked flags when using clang

jesschimein commented:
/AzurePipelines run

jesschimein commented:
/AzurePipelines run
